fundingmanager.go

```diff
@@ -393,8 +393,38 @@ func (f *fundingManager) Start() error {
 			case <-f.quit:
 				// The fundingManager is shutting down, and will
 				// resume wait on startup.
-			case <-doneChan:
+			case shortChanID, ok := <-confChan:
+				if !ok {
+					fndgLog.Errorf("waiting for funding" +
+						"confirmation failed")
+					return
+				}
 				// Success, funding transaction was confirmed.
+
+				// With the channel marked open, we'll create the
+				// state-machine object which wraps the database
+				// state.
+				lnChannel, err := lnwallet.NewLightningChannel(
```

Comment:
I know this is only part 1 of a two-part change, but this chunk of code is now repeated 4 times. I think it makes more sense to keep it in its own method for now, and the method name `sendFundingLockedAndAnnounceChannel` is still appropriate.

Reply:
I agree with all of the suggested changes besides `sendFundingLockedAndAnnounceChannel` because my later PR will have complicated logic that is better off from having them decoupled. There is a branch of code which ONLY sends the announcement, [but not the funding locked message](https://github.com/lightningnetwork/lnd/blob/master/fundingmanager.go#L446).  Additionally, the issue says to include adding to the channel router graph immediately after
sending the funding locked message. I think the code is cleaner having `sendFundingLocked`, `sendChannelAnnouncement`, & `addToRouterGraph` as separate functions.

Reply:
I'm not suggesting you remove the `sendFundingLocked`, `sendChannelAnnouncement`, and `addToRouterGraph` methods, I'm just suggesting you add another method `sendFundingLockedAndAnnounceChannel` (or `handleFundingConfirmation`), that calls those methods in the right order. I read over https://github.com/lightningnetwork/lnd/pull/332, and it appears the duplicated code exists there as well with no differences between the 4 occurrences.

Reply:
Ah gotcha.

Reply:
I like `handleFundingConfirmation` :)